examples/upload_videos/conservator_upload.py

```diff
@@ -0,0 +1,956 @@
+            except KeyError:
+                pass
+        has_updates = True
+        _logger.debug("       Metadata needs to be updated because of: tags")
+
+    upload_custom = default_metadata.metadata[camera_name_or_spectrum].get("custom")
+    if isinstance(upload_custom, dict):
+        existing_custom = video_metadata.get("custom")
+
+        if existing_custom is None:
+            # Just copy the whole document
+            has_updates = True
+            video_metadata["custom"] = upload_custom
+            _logger.debug("       Metadata needs to be updated because of: custom")
+        else:
+            # Selective update
+            if _update_custom_video_helper(
+                video_metadata, existing_custom, upload_custom
+            ):
+                has_updates = True
+                _logger.debug("       Metadata needs to be updated because of: custom")
+
+    if has_updates:
+        with open(save_path, "w", encoding="UTF-8") as json_file:
+            json.dump(
+                video_metadata_all,
+                json_file,
+                sort_keys=False,
+                indent=1,
+                separators=(",", ": "),
+            )
+
+        if dry_run is False:
+            video.upload_metadata(save_path)
+            stats.upload_metadata_count += 1
+            _logger.info("       Successfully updated video metadata")
+    else:
+        if dry_run is False:
+            _logger.info("       Video does not require metadata updates")
+
+    return has_updates
+
+
+def upload_video_helper(local_path: str, remote_name: str, collection: Collection):
+    """
+
+    :param local_path: The local file path to upload.
+
+    :param remote_name: If given, the remote name of the media. Otherwise, the local file
+        name is used.
+
+    :param collection: If specified, the Collection object, or `str` Collection ID to
+        upload the media to
+
+    :return: str of the video on success. None on failure
+    """
+
+    video_id = _conservator.videos.upload(
+        local_path, collection=collection, remote_name=remote_name
+    )
+    if not video_id:
+        _logger.error("Upload failed")
+
+    # You can wait for processing in this manner (not required).
+    # This script is designed to run multiples times
+    # print("Waiting for processing")
+    # _conservator.videos.wait_for_processing(video_id)
+    return video_id
+
+
+def get_video_and_collection(filename: str, conservator_location: str):
+
+    video = None
+
+    try:
+        # Get collection
+        collection = _conservator.collections.from_remote_path(
+            conservator_location, make_if_no_exist=False, fields="id"
+        )
+    except FLIR.conservator.wrappers.collection.InvalidRemotePathException:
+        collection = None
+
+    if collection is not None:
+        for video in collection.recursively_get_videos(
+            fields=["name", "owner"], search_text=f'name:"{filename}"'
+        ):
+            video: Video
```

Comment:
Is this an assignment?  It kind of looks like an out-of-place dict initializer.

Reply:
Looks like type hint for variable, which I think it sets a minimum python version of python3.9 for such code. Probably that minimum requirement was not intended, since conservator-cli is still advertising python 3.6 support in setup.py. Even if 3.6 is deemed too old, I think 3.8 is still reasonable to support.

Reply:
This script came from @filitchp, and I didn't do much more than lint/format and test it. I don't think we want to make a minimum mandatory version of Python (other than 3.x+), so I'll remove that.